[primTorch] Add a ref for `addcmul`

nkaretnikov opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #86731

Based on:
https://github.com/pytorch/pytorch/pull/79827
https://github.com/pytorch/pytorch/pull/72949

nkaretnikov committed:
[primTorch] Add a ref for `addcmul`

Based on:
https://github.com/pytorch/pytorch/pull/79827
https://github.com/pytorch/pytorch/pull/72949

[ghstack-poisoned]

nkaretnikov commented:
Note: I've made a tiny change to the `OpInfo` to keep this self-contained. LMK if you want all the suggested changes (C++ and error inputs) from https://github.com/pytorch/pytorch/pull/72949 as part of this PR.

lezcano commented on torch/_refs/__init__.py:
```diff
@@ -1582,6 +1583,36 @@ def addcdiv(
+    if value is not None:
+        dtype = self.dtype  # no scalars allowed, see add
+        python_type = utils.dtype_to_type(dtype)
+        if not utils.is_weakly_lesser_type(type(value), python_type):
+            msg = (
+                "value argument of type {0} cannot be safely cast to type {1}!".format(
+                    type(value), python_type
+                )
+            )
+            raise ValueError(msg)
```
Better to use a `utils.check`.

lezcano reviewed (approved):
One small comment, otherwise LGTM!

mruberry reviewed (approved):
See @lezcano's note about using `utils.check()`, otherwise LGTM!

nkaretnikov committed:
Update on "[primTorch] Add a ref for `addcmul`"


Based on:
https://github.com/pytorch/pytorch/pull/79827
https://github.com/pytorch/pytorch/pull/72949

[ghstack-poisoned]

nkaretnikov committed:
Update on "[primTorch] Add a ref for `addcmul`"


Based on:
https://github.com/pytorch/pytorch/pull/79827
https://github.com/pytorch/pytorch/pull/72949

[ghstack-poisoned]

nkaretnikov commented:
`addcmul`  failures from  https://github.com/pytorch/pytorch/issues/76047 no longer happen

nkaretnikov commented:
@pytorchbot merge -g

nkaretnikov committed:
Update on "[primTorch] Add a ref for `addcmul`"


Based on:
https://github.com/pytorch/pytorch/pull/79827
https://github.com/pytorch/pytorch/pull/72949

[ghstack-poisoned]

nkaretnikov commented:
@pytorchbot merge -g

nkaretnikov commented:
@pytorchbot merge -f 'circleci failures look unrelated'
